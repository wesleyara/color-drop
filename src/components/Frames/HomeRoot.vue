<script setup lang="ts"></script>

<template>
  <main
    class="mx-auto flex max-w-[1100px] flex-col items-center justify-center gap-5"
  >
    <slot></slot>
  </main>
</template>
