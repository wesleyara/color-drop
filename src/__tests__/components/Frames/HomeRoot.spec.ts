/* eslint-disable @typescript-eslint/no-explicit-any */
import { describe, it, expect, beforeEach } from "vitest";
import { mount, VueWrapper } from "@vue/test-utils";
import HomeRoot from "../../../components/Frames/HomeRoot.vue";

describe("HomeRoot unit tests", async () => {
  let sut: VueWrapper<any>;

  beforeEach(async () => {
    sut = mount(HomeRoot);
  });

  it("should render the component", () => {
    const wrapper = sut.exists();

    expect(wrapper).toBeDefined();
  });

  it("should match the snapshot", () => {
    const wrapper = sut.html();

    expect(wrapper).toMatchSnapshot();
  });

  it("should have a class", () => {
    const wrapper = sut.classes();

    expect(wrapper).toContain("flex");
  });
});
