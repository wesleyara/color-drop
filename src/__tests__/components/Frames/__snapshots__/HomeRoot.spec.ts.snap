// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeRoot unit tests > should match the snapshot 1`] = `"<main class="mx-auto flex max-w-[1100px] flex-col items-center justify-center gap-5"></main>"`;
